tools/cli/factory.go

```diff
@@ -238,7 +238,9 @@ func (b *clientFactory) newClientDispatcher(c *cli.Context, hostPortOverride str
 	})
 
 	if err := dispatcher.Start(); err != nil {
-		dispatcher.Stop()
+		if err := dispatcher.Stop(); err != nil {
+			ErrorAndExit("Failed to stop outbound transport channel", err)
+		}
 		b.logger.Fatal("Failed to create outbound transport channel: %v", zap.Error(err))
```

Comment:
probably better to use the logger for both here.  or update both to use ErrorAndExit if it looks better.

but generally I think we should use a logger if it exists, and leave printing as the bad-fallback-we-should-fix.

Reply:
got it. I'll use logger here